app/src/main/java/com/example/vybe/SearchProfilesActivity.java

```diff
@@ -0,0 +1,105 @@
+package com.example.vybe;
+
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.ListView;
+import android.widget.SearchView;
+import android.widget.Toast;
+
+import com.google.firebase.firestore.CollectionReference;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.FirebaseFirestoreException;
+import com.google.firebase.firestore.Query;
+import com.google.firebase.firestore.QueryDocumentSnapshot;
+import com.google.firebase.firestore.QuerySnapshot;
+
+import java.util.ArrayList;
+
+/**
+ * SearchProfilesActivity enables the user to search for other participants/users of the Vybe app
+ */
+public class SearchProfilesActivity extends AppCompatActivity {
+
+    private static final String TAG = "SearchProfilesActivity";
+    private FirebaseFirestore db = FirebaseFirestore.getInstance();
+
+    private ArrayList<User> usersList;
+    private ArrayList<User> searchList;
+    private ProfileAdapter profileAdapter;
+
+    private SearchView searchView;
+    private ListView searchListView;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_search_profiles);
+        Log.d(TAG, "onCreate: In search profiles");
+
+        searchView = findViewById(R.id.search_view);
+        searchListView = findViewById(R.id.search_list_view);
+
+        // Initialize list of users
+        usersList = new ArrayList<>();
+
+        // Get users from the database
+        final CollectionReference collectionReference = db.collection("Users");
+        Query query = collectionReference.orderBy("username", Query.Direction.DESCENDING);
+        query.addSnapshotListener((@Nullable QuerySnapshot queryDocumentSnapshots, @Nullable FirebaseFirestoreException e) -> {
+            usersList.clear();
+            for (QueryDocumentSnapshot doc : queryDocumentSnapshots) {
+                String username = (String) doc.getData().get("username");
+                String email = (String) doc.getData().get("email");
+                usersList.add(new User(username, email));   // Populate users list
```

Comment:
If we do decide to remove the user's personal profile from appearing in the search, we can compare email with the current user's email and if they match to not add them to this usersList